scene/3d/node_3d.cpp

```diff
@@ -597,33 +593,34 @@ void Node3D::_propagate_visibility_changed() {
 }
 
 void Node3D::show() {
-	if (data.visible) {
-		return;
-	}
-
-	data.visible = true;
+	_set_visible(true);
+}
 
-	if (!is_inside_tree()) {
-		return;
-	}
+void Node3D::hide() {
+	_set_visible(false);
+}
 
-	_propagate_visibility_changed();
+void Node3D::set_visible(bool p_visible) {
+	_set_visible(p_visible);
 }
```

Comment:
I don't see a difference between `set_visible` and `_set_visible`, so `_set_visible` should probably be removed?

Reply:
The only difference is one is public, the other is private. I generally dislike calling public methods from within the class, but it could be removed.

Reply:
I think calling public methods from other public methods is fine and pretty common in the Godot codebase.

Reply:
Alright, fixed it.